### ISSUE TITLE
Asset bundles don't parse correctly

### ISSUE THREAD

**Issue by Kirkman** [Kirkman/STTApi#1]
There still seems to be a problem with these assets, which started several months ago. 

As far as I can tell parseAssetBundle() doesn't parse `.sd`, `.ld`, and other asset bundle files correctly any more.

I don't know anything about Unity. Not sure if the `unitiyfs-asset-parser` library can be fixed, or if a new library is needed.

[This issue in an unrelated repo](https://github.com/HearthSim/UnityPack/pull/94) indicates the asset bundle format changed in 2019. Not sure if that's what we're facing here.

Also, another repo has its [own altered version](https://github.com/dragonjet/himegarip/tree/master/.overrides/unitiyfs-asset-parser) of `unitiyfs-asset-parser` from 8 months ago. Not sure if that bears on this problem, either.

**Comment by michaelsmoody** [comment]
Good find. I suspect that the format change _IS_ the issue. Looking into it now

**Comment by michaelsmoody** [comment]
Research notes:

https://www.imbushuo.net/blog/archives/505
https://github.com/marcan/deresuteme/commit/f388056646315b8cf33d2ef45715ebcd30fe261a
https://github.com/HearthSim/UnityPack and https://github.com/HearthSim/UnityPack/pull/94
https://github.com/keichi/binary-parser

More as I continue to research

**Comment by Kirkman** [comment]
Just a note to say I have been methodically going through `unitiyfs-asset-parser` and comparing it with HearthSim's `UnityPack` (which it clearly is based on). 

I made several changes that seem to be allowing `unitiyfs-asset-parser` to correctly parse v17 bundles (which is what I'm seeing from STT right now). 

It's still not quite working right in the app, though. I'll check in again soon.

**Comment by michaelsmoody** [comment]
Do you have a branch you're working on that I can pull and review?

**Comment by Kirkman** [comment]
No, I don't. It's still a mess, but I'll put something up here when I have a little more clarity.

**Comment by Kirkman** [comment]
It *is* working. I need to do a little more to properly integrate this updated version into STT, but you can see the changes here:

https://github.com/Kirkman/unityfs-asset-parser/commit/43a0da893a61eb6b761539c1c466d74bca2c46ce
